CalibTracker/SiStripHitEfficiency/plugins/HitEff.cc

```diff
@@ -592,7 +613,56 @@ void HitEff::analyze(const edm::Event& e, const edm::EventSetup& es) {
             }
           }
 
-          if (!tmpTmeas.empty()) {
+          //Test for two consecutive missing hits
+          if (int(TKlayers) - int(prev_TKlayers) == -3) {
+            foundConsMissingHits = true;
+            const DetLayer* detlayer = itm->layer();
+            const LayerMeasurements layerMeasurements{*measurementTrackerHandle, *measurementTrackerEvent};
+            const TrajectoryStateOnSurface tsos = itm->updatedState();
+            std::vector<DetLayer::DetWithState> compatDets = detlayer->compatibleDets(tsos, *thePropagator, *estimator);
+
+            if (misLayer > k_LayersStart && misLayer < k_LayersAtTOBEnd && previousMisLayer > k_LayersStart &&
+                previousMisLayer < k_LayersAtTOBEnd) {  //Barrel case
+              std::vector<BarrelDetLayer const*> barrelTIBLayers =
+                  measurementTrackerHandle->geometricSearchTracker()->tibLayers();
+              std::vector<BarrelDetLayer const*> barrelTOBLayers =
+                  measurementTrackerHandle->geometricSearchTracker()->tobLayers();
+              if (misLayer > k_LayersStart && misLayer < k_LayersAtTIBEnd) {
```

Comment:
Also?
```suggestion
              if (misLayer > k_LayersStart && misLayer <= k_LayersAtTIBEnd) {
```

Reply:
The cases where the innermost missed layer (in case of 2 consecutive ones) is located in the last TOB layer is not taken into account (and should be very scarce), since the previous missed layer = missed layer + 1, this would reach outside the array size